### PR TITLE
fix(amplify_auth_cognito): deprecates globalSignOut; simple user check upon Android signIn

### PR DESCRIPTION
**Pull request by haverchuck**
*Description of changes:*

 Removes globalSignOut; simple user check upon Android signIn

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by fjnoyp — APPROVED** [review]

File: packages/amplify_auth_cognito/android/src/main/kotlin/com/amazonaws/amplify/amplify_auth_cognito/types/FlutterSignInRequest.kt [hunk 1]
```diff
@@ -38,5 +40,17 @@ data class FlutterSignInRequest(val map: HashMap<String, *>) {
         }
       }
     }
+    fun checkUser() {
+      try {
+        var user: AuthUser? = Amplify.Auth.currentUser;
+        if (user != null) {
+          throw FlutterInvalidStateException("There is already a user  signed in.", "Sign out before calling sign in.")
```
Comment (Ashish-Nanda) [inline packages/amplify_auth_cognito/android/src/main/kotlin/com/amazonaws/amplify/amplify_auth_cognito/types/FlutterSignInRequest.kt:47]: Nit: Is this the same language as iOS?
Shouldn't the recovery suggestion refer to the API name `signIn`:  
```
"Sign out before calling signIn()"
```

Comment (haverchuck) [reply packages/amplify_auth_cognito/android/src/main/kotlin/com/amazonaws/amplify/amplify_auth_cognito/types/FlutterSignInRequest.kt:47]: @Ashish-Nanda It is very close - it corrects a very small grammatical error.  And ios does not use "()".

Comment (haverchuck) [reply packages/amplify_auth_cognito/android/src/main/kotlin/com/amazonaws/amplify/amplify_auth_cognito/types/FlutterSignInRequest.kt:47]: And I think "()" is implied by "calling".

**Review by Ashish-Nanda — APPROVED** [review]
LGTM

**Pull request merged**